src/cache.c

```diff
@@ -256,15 +258,9 @@ hsk_cache_key_equal(const void *a, const void *b) {
 }
 
 bool
-hsk_cache_key_set(hsk_cache_key_t *ck, const char *name, uint16_t type) {
+hsk_cache_key_set(hsk_cache_key_t *ck, const uint8_t *name, uint16_t type) {
   assert(ck);
 
-  if (!hsk_dns_name_verify(name))
```

Comment:
Don't we still need to verify if `tld` is correct ?

Otherwise this method will never return false.
I guess that could make this more precise and instead move tld check to the `hsk_cache_insert_data` and `hsk_cache_get_data` before the cache_key_set.

Reply:
This method will become `void` and instead those two methods will return early.

Reply:
Good point, I think we can remove the return value. We check `hsk_dns_name_verify()` in `hsk_ns_onrecv()` which is the entry point for the user's request and is checked before attempting name resolution and before caching. So I think throughout hnsd we can assume that `req->tld` is already valid.

so I changed `hsk_cache_key_set()` to void -- what did you mean by "two methods" ?

fixed in f5200ab1251950b897e8ef3f71f5a11737aeaa5a